### PR TITLE
enable perl and shell linting for `opennms-repo`

### PR DESCRIPTION
**Pull request by RangerRick**
do some basic linting in CircleCI

### CODE REVIEW

**Comment by RangerRick** [comment]
this builds on my previous PR [here](https://github.com/OpenNMS/opennms-repo/pull/114) -- that should be merged first

File: script/deploy-debs.pl [hunk 1]
```diff
@@ -52,17 +50,19 @@
...
-while (my $line = readdir(FILES)) {
+my $FILEDIR_HANDLE;
+
+opendir($FILEDIR_HANDLE, '.') or die "Unable to read current directory: $!";
+while (my $line = readdir($FILEDIR_HANDLE)) {
 	next if ($line =~ /^\.\.?$/);
 	chomp($line);
 	if ($line =~ /\.deb$/) {
-		push(@FILES_DEBS, $line);
+		push(@$FILEDIR_HANDLE_DEBS, $line);
```
Comment (deejgregor) [inline script/deploy-debs.pl]: I think this was mistakenly changed.

File: script/deploy-rpms.pl [hunk 2]
```diff
@@ -67,37 +65,40 @@
...
+my $FILEDIR_HANDLE;
+opendir($FILEDIR_HANDLE, '<', '.') or die "Unable to read current directory: $!";
+while (my $line = readdir($FILEDIR_HANDLE)) {
 	next if ($line =~ /^\.\.?$/);
 	chomp($line);
 	if ($line =~ /^opennms-source-.*\.tar.(gz|bz2)$/) {
 		$FILE_SOURCE_TARBALL = $line;
 	} elsif ($line =~ /\.rpm$/) {
-		push(@FILES_RPMS, $line);
+		push(@$FILEDIR_HANDLE_RPMS, $line);
```
Comment (deejgregor) [inline script/deploy-rpms.pl]: I think this was mistakenly changed.

File: script/update-doc-repo.pl [hunk 3]
```diff
@@ -852,12 +866,13 @@ sub process_minion_asciidoc_docdir {
...
-	opendir(DIR, $docdir) or die "Failed to open $docdir for reading: $!\n";
-	for my $entry (sort { lc($a) cmp lc($b) } grep { /\.html$/ } readdir(DIR)) {
+	my $DOCDIR_HANDLE;
+	opendir($DOCDIR_HANDLE, $docdir) or die "Failed to open $docdir for reading: $!\n";
+	for my $entry (sort { lc($a) cmp lc($b) } grep { /\.html$/ } readdir($DOCDIR_HANDLE)) {
 		my ($name) = $entry =~ /^(.*?)\.html$/;
 		$name = lc($name);
 
-		my $target = File::Spec->catdir($INSTALLDIR, $name);
+		my $target = File::Spec->catdir($INSTALL$DOCDIR_HANDLE, $name);
```
Comment (deejgregor) [inline script/update-doc-repo.pl]: I think this was mistakenly changed.

File: script/update-doc-repo.pl [hunk 4]
```diff
@@ -922,13 +937,15 @@ sub process_docbook_docdir {
...
+	my $DIR_HANDLE;
+
+	opendir($DIR_HANDLE, $from) or die "Failed to open $from for reading: $!\n";
+	my @files = sort grep { /\.(pdf|html)$/ } readdir($DIR_HANDLE);
+	closedir($DIR_HANDLE) or die "Failed to close $from: $!\n";
 
 	my $common = File::Spec->catdir($from, 'common');
 	if (-d $common) {
-		my $tocommon = File::Spec->catdir($INSTALLDIR, '.common');
+		my $tocommon = File::Spec->catdir($INSTALL$DIR_HANDLE, '.common');
```
Comment (deejgregor) [inline script/update-doc-repo.pl]: I think this was mistakenly changed.

File: script/update-doc-repo.pl [hunk 5]
```diff
@@ -966,7 +983,7 @@ sub process_docbook_docdir {
 		do_debug("- Processing $name");
 
 		my $mappedname = (exists $mapping->{$name}? $mapping->{$name} : $name);
-		my $to = File::Spec->catdir($INSTALLDIR, $mappedname);
+		my $to = File::Spec->catdir($INSTALL$DIR_HANDLE, $mappedname);
```
Comment (deejgregor) [inline script/update-doc-repo.pl]: I think this was mistakenly changed.

**Review by deejgregor — CHANGES_REQUESTED** [review]
I think a few variables were mistakenly changed. See per-line comments.

File: script/deploy-debs.pl (hunk 1 above)
Comment (deejgregor) [reply script/deploy-debs.pl]: I went ahead and pushed the fix for this in 09221ba03061f827f11745fc76aa389094aefa11.

File: script/deploy-rpms.pl (hunk 2 above)
Comment (deejgregor) [reply script/deploy-rpms.pl]: I went ahead and pushed the fix for this in 09221ba03061f827f11745fc76aa389094aefa11.

File: script/update-doc-repo.pl (hunk 3 above)
Comment (deejgregor) [reply script/update-doc-repo.pl]: I went ahead and pushed the fix for this in 09221ba03061f827f11745fc76aa389094aefa11.

File: script/update-doc-repo.pl (hunk 5 above)
Comment (deejgregor) [reply script/update-doc-repo.pl]: I went ahead and pushed the fix for this in 09221ba03061f827f11745fc76aa389094aefa11.

**Review by deejgregor — APPROVED** [review]
I think this is ready to go!

I went ahead and pushed a few fixes in 09221ba03061f827f11745fc76aa389094aefa11.

**Comment by RangerRick** [comment]
Good catch, clearly this was a PR I pushed on a friday afternoon and didn't want to lose, but still needed to look over. ;)

**Pull request merged**